pkg/image/admission/imagestream_limits_test.go

```diff
@@ -0,0 +1,420 @@
+package admission
+
+import (
+	"strings"
+	"testing"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/client/cache"
+	fake "k8s.io/kubernetes/pkg/client/unversioned/testclient"
+
+	imagetest "github.com/openshift/origin/pkg/image/admission/testutil"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	quotautil "github.com/openshift/origin/pkg/quota/util"
+)
+
+func TestGetMaxLimits(t *testing.T) {
+	for _, tc := range []struct {
+		name           string
+		lrs            []kapi.LimitRange
+		expectedLimits kapi.ResourceList
+	}{
+		{
+			name: "no limit range",
+		},
+
+		{
+			name: "unrelevant limit range",
+			lrs: []kapi.LimitRange{
+				{
+					Spec: kapi.LimitRangeSpec{
+						Limits: []kapi.LimitRangeItem{
+							{
+								Type: kapi.LimitTypePod,
+								Max:  kapi.ResourceList{kapi.ResourceCPU: resource.MustParse("200m")},
+							},
+						},
+					},
+				},
+			},
+		},
```

Comment:
I think it would make the test unnecessary complex because several limit ranges need to be injected at once. Not event the listing of limits can be easily replaced with some factory because there are different combination of types and resources tested.